lib/schemas.js

```diff
@@ -38,6 +38,11 @@ Schemas.prototype.resolve = function (id) {
 }
 
 Schemas.prototype.resolveRefs = function (routeSchemas, dontClearId) {
+  // alias query to querystring schema
+  if (routeSchemas.query) {
+    routeSchemas.querystring = routeSchemas.query
```

Comment:
I think we should throw an error if the user set the `query` and the `querystring` fields

Reply:
@Eomm as mentioned here right? https://github.com/fastify/fastify/issues/1690#issuecomment-499879028

Reply:
I didn't saw it, but yes: it is the same, overriding without control could be confusing for the user for sure 👍

Reply:
I can agree with generating an error if both are supplied.

Reply:
Just curious, how about merging if both supplied?

Reply:
This could generate more errors because if you put the same JSON schema for both the fields and merge those, you will get an error during the validation due to the duplicated $id.

So, for the user, it would be really hard to build 2 JSON schema that can merge to one valid JSON Schema and it doesn't worth the effort to implement a merge function

Reply:
Will update the PR accordingly.

Reply:
PR updated! Hope this approach's fine!